app/services/AdjacentCellsCalculator.coffee

```diff
@@ -11,15 +12,16 @@ class AdjacentCellsCalculator
       for j in [@x - 1, @x, @x + 1]
         continue if i is @y and j is @x
         @cells.push @validLocation @grid, i, j
-    @cells
+    @cells.set
 
   # returns a valid location if it exists, otherwise null
   validLocation: (grid,x,y) =>
     while grid.validIndices x, y
-      return new Tuple x, y if @empty grid, x, y
+      return new Tuple x, y if (@empty grid, x, y)
       y--
     null
 
-  empty: (grid,x,y) => grid.at x, y is null
+  empty: (grid,x,y) =>
+    (grid.at x, y) is null
```

Comment:
I propose we change `grid.at` to return ' ' (SPACE character) to indicate empty and null to indicate invalid index.